kernel/mm/system_mmap.c

```diff
@@ -538,7 +539,9 @@ linear_map_mem_region(struct mem_region *r, ulong *vbegin, ulong *vend)
    const ulong pend = MIN((ulong)(r->addr+r->len), (ulong)LINEAR_MAPPING_SIZE);
 
    const bool rw = (r->type == MULTIBOOT_MEMORY_AVAILABLE) ||
-                   (r->extra & MEM_REG_EXTRA_KERNEL);
+                   (r->extra & MEM_REG_EXTRA_KERNEL) ||
+                   (r->type == EFI_RUNTIME_CODE) ||
```

Comment:
We don't need to map the EFI_RUNTIME_CODE in read/write mode. Just the data. The code should be mapped read-only.